File: back-go/domain/services/strava/segment.go
package strava

type Segment struct {
	ActivityType  string    `json:"activity_type"`
	AverageGrade  float64   `json:"average_grade"`
	City          *string   `json:"city,omitempty"`
	ClimbCategory int       `json:"climb_category"`
	Country       *string   `json:"country,omitempty"`
	Distance      float64   `json:"distance"`
	ElevationHigh float64   `json:"elevation_high"`
	ElevationLow  float64   `json:"elevation_low"`
	EndLatLng     []float64 `json:"end_latlng"`
	Hazardous     bool      `json:"hazardous"`
	Id            int64     `json:"id"`
	MaximumGrade  float64   `json:"maximum_grade"`
	Name          string    `json:"name"`
	IsPrivate     bool      `json:"private"`
	ResourceState int       `json:"resource_state"`
	Starred       bool      `json:"starred"`
	StartLatLng   []float64 `json:"start_latlng"`
	State         *string   `json:"state,omitempty"`
}

File: back-kotlin/src/main/kotlin/me/nicolas/stravastats/domain/business/strava/Segment.kt
package me.nicolas.stravastats.domain.business.strava

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class Segment(
    @param:JsonProperty("activity_type")
    val activityType: String,
    @param:JsonProperty("average_grade")
    val averageGrade: Double,
    val city: String?,
    @param:JsonProperty("climb_category")
    val climbCategory: Int,
    val country: String?,
    val distance: Double,
    @param:JsonProperty("elevation_high")
    val elevationHigh: Double,
    @param:JsonProperty("elevation_low")
    val elevationLow: Double,
    @param:JsonProperty("end_latlng")
    val endLatLng: List<Double>,
    val hazardous: Boolean,
    val id: Long,
    @param:JsonProperty("maximum_grade")
    val maximumGrade: Double,
    val name: String,
    @param:JsonProperty("private")
    val isPrivate: Boolean,
    @param:JsonProperty("resource_state")
    val resourceState: Int,
    val starred: Boolean,
    @param:JsonProperty("start_latlng")
    val startLatLng: List<Double>,
    val state: String?,
)